contracts/core/UniversalLiquidatorRegistry.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+// imported contracts and libraries
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+// interfaces
+import "../interface/IUniversalLiquidatorRegistry.sol";
+
+// libraries
+import "../libraries/DataTypes.sol";
+import "../libraries/Errors.sol";
+
+// constants and types
+import {ULRegistryStorage} from "./storage/ULRegistry.sol";
+
+contract UniversalLiquidatorRegistry is
+    Ownable,
+    IUniversalLiquidatorRegistry,
+    ULRegistryStorage
+{
+    function getPath(
+        address _sellToken,
+        address _buyToken
+    ) public view override returns (DataTypes.SwapInfo[] memory) {
+        if (paths[_sellToken][_buyToken].dex != bytes32(0)) {
+            DataTypes.SwapInfo[] memory retPaths = new DataTypes.SwapInfo[](1);
+            retPaths[0] = DataTypes.SwapInfo(
+                dexesInfo[paths[_sellToken][_buyToken].dex],
+                paths[_sellToken][_buyToken].paths
+            );
+            return retPaths;
+        }
+
+        for (uint256 idx; idx < intermediateTokens.length; ) {
+            if (
+                paths[_sellToken][intermediateTokens[idx]].dex != bytes32(0) &&
+                paths[intermediateTokens[idx]][_buyToken].dex != bytes32(0)
+            ) {
+                // found the intermediateToken and intermediateDex
+                DataTypes.SwapInfo[] memory retPaths = new DataTypes.SwapInfo[](
+                    2
+                );
+                retPaths[0] = DataTypes.SwapInfo(
+                    dexesInfo[paths[_sellToken][intermediateTokens[idx]].dex],
+                    paths[_sellToken][intermediateTokens[idx]].paths
+                );
+                retPaths[1] = DataTypes.SwapInfo(
+                    dexesInfo[paths[intermediateTokens[idx]][_buyToken].dex],
+                    paths[intermediateTokens[idx]][_buyToken].paths
+                );
+                return retPaths;
+            }
+            unchecked {
+                ++idx;
+            }
+        }
+        revert("Liquidation path is not set");
+    }
+
+    function setPath(

```

Comment:
This can be simplified by not taking `_sellToken` and `_buyToken` as inputs, and just extracting them from the `_paths` input.

Reply:
Also, it should be checked that `_dex` is a valid dex.

Reply:
Fixed!